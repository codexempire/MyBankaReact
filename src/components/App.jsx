/* eslint-disable import/no-cycle */
import React, { Fragment } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Routes from './route';
import store from '../store/store';

function App() {
  // const username = 'michael';
  return (
    <Provider store={store}>
      <Router>
        <Fragment>
          <Routes />
        </Fragment>
      </Router>
    </Provider>
  );
}

export default App;
