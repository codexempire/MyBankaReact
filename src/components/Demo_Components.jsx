import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export function User(props) {
  const { username } = props;

  const button = () => (
    <Link to="/">
      <button type="button" className="homeButton">
        To Home
      </button>
    </Link>
  );
  if (username !== '') {
    return (
      <React.Fragment>
        <div className="user logged">
          Welcome to the users route
          {' '}
          {username}
        </div>
        {button()}
      </React.Fragment>
    );
  }
  return (
    <React.Fragment>
      <div className="user unlogged">This is the user route</div>
      {button()}
    </React.Fragment>
  );
}
User.propTypes = {
  username: PropTypes.string,
};

User.defaultProps = {
  username: '',
};
export function Home() {
  return (
    <React.Fragment>
      <div className="home">Here is the home route</div>
    </React.Fragment>
  );
}
