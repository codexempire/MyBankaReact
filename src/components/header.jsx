import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <Fragment>
    <h1>Hi React</h1>
    <ul>
      <li><Link to="/">Home</Link></li>
      <li><Link to="/user">Users</Link></li>
    </ul>
  </Fragment>
);

export default Header;
