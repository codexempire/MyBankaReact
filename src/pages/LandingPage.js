import React from 'react';
import { Link } from 'react-router-dom';
import { marginRight } from '../components/styles'

export default () => (
  <>
    <header className="grid head">
      <div className="image" />
      <div className="container">
        <h1>
          <strong>M</strong>
          y
          <strong>B</strong>
          anka
        </h1>
        <p> A bank you can trust...</p>
        <Link to="/login" className="btn btn-purple" style={marginRight}>Login</Link>
        <Link to="/signup" className="btn btn-border-y">Signup</Link>
      </div>
    </header>
  </>
);