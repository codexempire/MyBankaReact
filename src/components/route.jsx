/* eslint-disable import/no-cycle */
import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { User, Home } from './Demo_Components';

const Routes = () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route exact path="/user" component={User} />
  </Switch>
);

export default Routes;
