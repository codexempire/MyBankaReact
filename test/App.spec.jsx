import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from './enzyme';
import { User, Home } from '../src/components/Demo_Components';
import App from '../src/components/App';
import Routes from '../src/components/route';
import Header from '../src/components/header';
import LandingPage from '../src/pages';

describe('User tests', () => {
  it('renders User page with a username', () => {
    const username = 'codexempire';
    const expectedString = `Welcome to the users route ${username}`;
    const wrapper = shallow(<User username={username} />);

    expect(wrapper.find('.user').text()).toEqual(expectedString);
    expect(wrapper.find('.user').hasClass('logged')).toEqual(true);
    expect(wrapper.find('.homeButton').props().children).toEqual('To Home');
  });

  it('renders a User page without a logged user', () => {
    const expectedString = 'This is the user route';
    const wrapper = shallow(<User />);

    expect(wrapper.find('.user').text()).toEqual(expectedString);
    expect(wrapper.find('.user').hasClass('unlogged')).toEqual(true);
    expect(wrapper.find('.homeButton').props().children).toEqual('To Home');
  });
});

describe('Home tests', () => {
  it('renders applications Home page', () => {
    const expectedString = 'Here is the home route';
    const wrapper = shallow(<Home />);

    expect(wrapper.props().children.props.className).toEqual('home');
    expect(wrapper.find('.home').text()).toEqual(expectedString);
  });
});

describe('App', () => {
  it('should render the App component correctly', () => {
    const shallowWrapper = shallow(<App><User /></App>);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('User')).toBeTruthy();
  });
});

describe('Routes', () => {
  it('should render the routes for the application correctly', () => {
    const shallowWrapper = shallow(<Routes />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});

describe('Header', () => {
  it('should render the header for the application correctly', () => {
    const shallowWrapper = shallow(<Header />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});

describe('Landing page', () => {
  it('shoul render the landing page for the application correctly', () => {
    const shallowWrapper = shallow(<LandingPage />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
