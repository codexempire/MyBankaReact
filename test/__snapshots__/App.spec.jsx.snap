// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Header />
    <Routes />
  </BrowserRouter>
</Provider>
`;

exports[`Header should render the header for the application correctly 1`] = `
<Fragment>
  <h1>
    Hi React
  </h1>
  <ul>
    <li>
      <Link
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        to="/user"
      >
        Users
      </Link>
    </li>
  </ul>
</Fragment>
`;

exports[`Routes should render the routes for the application correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/user"
  />
</Switch>
`;
